### DISCUSSION TITLE
Suggestion: Fail Louder

### DISCUSSION THREAD

**Discussion by ExeArco** [Chatterino/chatterino2#3213] [Ideas]
Whenever your twitch login expires it writes cleanly and neatly in chat that your session has expired and needs to be renewed.
This is a painless process that takes 10 seconds and I have no problem with as there is nothing that can be really done about limits of Twitch's api.
However, what I do have a problem with is how quietly it fails.
The tiny little notification always gets scrolled up rather quickly if youre in any other chat and without renewing your session many other features don't work.
For example, the streamer has gone live notifications don't work if you haven't renewed your session.
I missed a large chunk of a streamer I wanted to watch due to this.
I think a good way around this would be turning the user's text red (in the title bar, next to the Chatterino and the gear) when it needs to be renewed, as this would be easily and quickly noticeable without interrupting anything else on the user's computer.

**Comment by fourtf** [comment]
A better way to handle this is to renew the fucking token in code

**Comment by Felanbird** [comment]
I agree with fourtf, this will be better handled in #3065, but forcing Chatterino to check for you if your token is going to expire soon, I feel like this just would be.... odd.

If want to manually keep up with when your token will expire, this information can easily be checked via 3rd-party tools:
https://codepen.io/Alca/full/BaLrORm - this will give you the exact time until your token will expire.
![image](https://user-images.githubusercontent.com/41973452/131467162-1ad353e5-64ef-4238-89f2-4966c0ffde7b.png)
oauthToken can be found in the `settings.json` file `C:\Users\$USER\AppData\Roaming\chatterino2\settings\settings.json` [windows]
![image](https://user-images.githubusercontent.com/41973452/131467565-958dfccf-fe30-46a9-85eb-bac43cb29e0a.png)
or can be pulled directly from the box on the client_login page.
![image](https://user-images.githubusercontent.com/41973452/131469283-a95b5664-b1f1-4643-ab4a-6aab596ba5c2.png)
